application/constants/templates.py

```diff
@@ -43,3 +53,62 @@ class HookTypes(StrEnum):
     Template application hook types.
     """
     kubernetes_job = 'kubernetes_job'
+
+
+class ComponentTypeHttpMethods(StrEnum):
+    """
+    Template HTTP component methods.
+    """
+    get = 'get'
+    post = 'post'
+    put = 'put'
+    patch = 'patch'
+    delete = 'delete'
+
+
+class HttpDeploy(BaseModel):
+    """
+    Deploy schema for HTTP type components.
+    """
+    url: HttpUrl = Field(description='URL for HTTP deployment.', example='http://example.com/path/to/deploy')
+    method: ComponentTypeHttpMethods = Field(
+        default=ComponentTypeHttpMethods.get,
+        description='HTTP method for deployment',
+        example=ComponentTypeHttpMethods.post)
+    headers: dict | list[dict] | None = Field(
+        description='Headers for HTTP deployment', example={
+            'Content-Type': 'application/json'})
+    parameters: dict | list[dict] | None = Field(
+        description='Parameters for HTTP deployment', example=[{'param1': 'value1'}])
+
+
+class HttpStatus(BaseModel):
+    """
+    Status schema for HTTP type components.
+    """
+    url: HttpUrl = Field(description='URL for HTTP status.', example='http://example.com/path/to/status')
+    method: ComponentTypeHttpMethods = Field(
+        default=ComponentTypeHttpMethods.get,
+        description='HTTP method for status',
+        example=ComponentTypeHttpMethods.post)
+    headers: dict | list[dict] | None = Field(
+        description='Headers for HTTP status', example={
+            'Content-Type': 'application/json'})
+    parameters: dict | list[dict] | None = Field(
+        description='Parameters for HTTP status', example=[{'param1': 'value1'}])
+
+
+class HttpDelete(BaseModel):
+    """
+    Delete schema for HTTP type components.
+    """
+    url: HttpUrl = Field(description='URL for HTTP delete.', example='http://example.com/path/to/delete')
+    method: ComponentTypeHttpMethods = Field(
```

Comment:
I'm not sure like 100% but please consider replacing `ComponentTypeHttpMethods` by `HTTPMethods` from `application/constants/common.py`.

Reply:
I would like to have controll over the supported methods of ComponentTypeHttpMethods. ie  not all methods in HTTPMethods should be in HTTPMethods.